common/src/main/java/stirling/software/common/util/EmlToPdf.java

```diff
@@ -0,0 +1,1522 @@
+            html.append("<div class=\"attachment-section\">\n");
+            int displayedAttachmentCount =
+                    content.getAttachmentCount() > 0
+                            ? content.getAttachmentCount()
+                            : content.getAttachments().size();
+            html.append("<h3>Attachments (").append(displayedAttachmentCount).append(")</h3>\n");
+
+            if (!content.getAttachments().isEmpty()) {
+                for (EmailAttachment attachment : content.getAttachments()) {
+                    // Create attachment info with paperclip emoji before filename
+                    String uniqueId =
+                            "attachment_"
+                                    + attachment.getFilename().hashCode()
+                                    + "_"
+                                    + System.nanoTime();
+                    attachment.setEmbeddedFilename(
+                            attachment.getEmbeddedFilename() != null
+                                    ? attachment.getEmbeddedFilename()
+                                    : attachment.getFilename());
+
+                    html.append("<div class=\"attachment-item\" id=\"").append(uniqueId).append("\">")
+                            .append("<span class=\"attachment-icon\">").append("\uD83D\uDCCE").append("</span> ")
+                            .append("<span class=\"attachment-name\">")
+                            .append(escapeHtml(safeMimeDecode(attachment.getFilename())))
+                            .append("</span>");
+
+                    String sizeStr = formatFileSize(attachment.getSizeBytes());
+                    html.append(" <span class=\"attachment-details\">(").append(sizeStr);
+                    if (attachment.getContentType() != null && !attachment.getContentType().isEmpty()) {
+                        html.append(", ").append(escapeHtml(attachment.getContentType()));
+                    }
+                    html.append(")</span></div>\n");
+                }
+            }
+
+            if (request.isIncludeAttachments()) {
+                html.append("<div class=\"attachment-info-note\">\n");
+                html.append(
+                        "<p><em>Attachments are embedded in the file.</em></p>\n");
+                html.append("</div>\n");
+            } else {
+                html.append("<div class=\"attachment-info-note\">\n");
+                html.append(
+                        "<p><em>Attachment information displayed - files not included in PDF.</em></p>\n");
+                html.append("</div>\n");
+            }
+
+            html.append("</div>\n");
+        }
+
+        html.append("</div>\n");
+        html.append("</body></html>");
+
+        return html.toString();
+    }
+
+    private static byte[] attachFilesToPdf(byte[] pdfBytes, List<EmailAttachment> attachments)
+            throws IOException {
+        try (PDDocument document = Loader.loadPDF(pdfBytes);
```

Comment:
Use CustomPDFDocumentFactory class to allow loading of large sized PDFs memory efficiently